### ISSUE TITLE
CORS package not working correctly

### ISSUE THREAD

**Issue by Guergeiro** [honojs/hono#828]
Since version v2.5.7, the cors package of Hono stopped doing its work. Version v2.5.6, the latest working version, had the following changes applied that are relevant to the issue: #792 #693 

**Platform:** Cloudflare Workers using Wrangler v2.8.0 (localhost testing).

**Steps to reproduce (v2.5.6 and v2.7.3)**

1. Create a simple backend that uses cors package (default `*`).
2. Run the server.
3. Try to fetch from a different client.

**Expected behaviour:** Fetching from a different client should happen normally.
**Actual behaviour:** CORS errors even though cors package is used.

**Comment by yusukebe** [comment]
Hi @Guergeiro !

Thank you for creating the issue. But I can't reproduce it in my code below. It works well.

<https://gist.github.com/yusukebe/22e630581092dcc0d9d1f6ce0763d329>

I can fix/investigate it if it's reproduced.

**Comment by Guergeiro** [comment]
So, I was able to reproduce this by having the following Hono code:
```ts
import { Hono } from 'hono'
import { cors } from 'hono/cors'
const app = new Hono()

app.use(cors())
app.get('/', (c) => new Response(JSON.stringify({heellow: "world"})))

export default app
```

This used to work before, as I never used the Hono's wrapper methods and the function signature always expected a Response.

**Comment by yusukebe** [comment]
It was a bug related to handling `Response`. I've fixed it and shipped v2.7.4. I believe it works correctly with CORS middleware.

Thank you very much for creating the issue!
